### PR TITLE
Make theme selector

### PR DESCRIPTION
**Pull request by glebdronchenko**

### CODE REVIEW

File: src/utils/prepareData.js [hunk 1]
```diff
@@ -12,4 +12,17 @@ function prepareChartData(hourlyForecast) {
   return [hours, temperatures];
 }
 
-export { prepareChartData };
+function prepareThemesList(themeConfig) {
```
Comment (anton-vlasik) [inline src/utils/prepareData.js]: If you already using lodash - there is 'lodash/values' I think

File: src/components/ThemeSelect/components/ThemeDropDown/ThemeDropDown.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import ThemeItem from '../ThemeItem/ThemeItem';
+import classes from './ThemeDropDown.module.scss';
+import themeConfig from '../../../../config/themeConfig';
+import { prepareThemesList } from '../../../../utils/prepareData';
+
+function ThemeDropDown() {
+  const themeList = prepareThemesList(themeConfig);
```
Comment (anton-vlasik) [inline src/components/ThemeSelect/components/ThemeDropDown/ThemeDropDown.jsx]: Again: this is out of scope of ThemeDropDown. Apparently you can do that once in your app. Or even change your theme object structure to avoid taking values of themeList object

File: src/components/ThemeSelect/components/ThemeItem/ThemeItem.jsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import classes from './ThemeItem.module.scss';
+import ThemeContext from '../../../../providers/themeContext';
+
+function ThemeItem({ theme }) {
+  return (
+    <ThemeContext.Consumer>
```
Comment (anton-vlasik) [inline src/components/ThemeSelect/components/ThemeItem/ThemeItem.jsx]: Are you sure that Context.Consumer should wrap each individual Item but not the whole list of items instead?

File: src/hooks/commonHooks.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick, true);
+
+    return () => {
+      document.removeEventListener('click', handleClick, true);
+    };
+  }, [refElement]);
```
Comment (anton-vlasik) [inline src/hooks/commonHooks.js]: why do you use refElement as a dependency?

File: src/hooks/commonHooks.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { useEffect, useRef, useState } from 'react';
+
+export function useDropdownVisible(initialVisible) {
```
Comment (anton-vlasik) [inline src/hooks/commonHooks.js]: Seems like you want to check outside clicks of your element. It is not a matter of visibility  - you are inspecting if click was outside or inside. Moreover is it only working with dropdowns? So it seems to me that the name is not quite good enough. Either you work only with dropdowns or it is more like useClickoutside. - but then it is up to consuming component to do something when click was outside. Task of this hook would just to tell the outter world that it was registered outside passed element,

File: src/utils/prepareData.js (hunk 1 above)
Comment (glebdronchenko) [reply src/utils/prepareData.js]: No, I don't use lodash.

File: src/hooks/commonHooks.js (hunk 4 above)
Comment (glebdronchenko) [reply src/hooks/commonHooks.js]: No. It's my mistake. It doesn't have to be here. At first I wanted to provide this variable to hook as function parameter and I thought, that this parameter could change but I was wrong.

File: src/hooks/commonHooks.js (hunk 5 above)
Comment (glebdronchenko) [reply src/hooks/commonHooks.js]: I agree it isn't correct hook name.

File: src/components/ThemeSelect/components/ThemeDropDown/ThemeDropDown.jsx (hunk 2 above)
Comment (glebdronchenko) [reply src/components/ThemeSelect/components/ThemeDropDown/ThemeDropDown.jsx]: I have moved all theme config methods to theme config.

File: src/components/ThemeSelect/components/ThemeItem/ThemeItem.jsx (hunk 3 above)
Comment (glebdronchenko) [reply src/components/ThemeSelect/components/ThemeItem/ThemeItem.jsx]: I have removed this theme context and pass context values ​​from parent component.

File: src/components/ThemeSelect/components/ThemeDropDown/ThemeDropDown.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import ThemeItem from '../ThemeItem/ThemeItem';
+import classes from './ThemeDropDown.module.scss';
+import { getThemesList } from '../../../../config/themeConfig';
+import ThemeContext from '../../../../providers/themeContext';
+
+function ThemeDropDown({ currentTheme, selectTheme }) {
+  const themeList = getThemesList();
```
Comment (anton-vlasik) [inline src/components/ThemeSelect/components/ThemeDropDown/ThemeDropDown.jsx]: Again: does it need to be in the scope of this component? HINT: no

Comment (glebdronchenko) [reply src/components/ThemeSelect/components/ThemeDropDown/ThemeDropDown.jsx]: I moved it outside of component scope.

**Review by ximet — APPROVED** [review]

**Review by anton-vlasik — APPROVED** [review]
